website/src/docs/react-initializing.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "Initializing Uppy"
+type: docs
+module: "@uppy/react"
+permalink: docs/react/initializing/
+alias: docs/react/initializing/
+order: 1
+category: "React"
+---
+
+Your Uppy instance must be initialized before passing it to an `uppy={}` prop in the desired component,
+and should be cleaned up using `uppy.close()` when you are done with it.
+
+## Class Components
+
+A simple approach is to initialize it in your React component's `constructor()` and destroy it in `componentWillUnmount()`.
+
+> ⚠ Uppy instances are stateful, so the same instance must be used across different renders.
+> Do **NOT** initialize Uppy in a `render()` method!
+
+```js
+class MyComponent extends React.Component {
+  constructor (props) {
+    super(props)
+    this.uppy = Uppy()
+      .use(Transloadit, {})
+  }
+
+  componentWillUnmount () {
+    this.uppy.close()
+  }
+
+  render () {
+    return <DashboardModal uppy={this.uppy} />
+  }
+}
+```
+
+## Functional Components
+
+With react hooks, writing pure functional components are on hype. For wealthy initializing Uppy in a functional component, we need to use the `useRef` hook for keeping the same instance across multiple renders and use `useEffect` for cleaning up things when the component is unmounted.
+
+```js
+const MyComponent = () => {
+  const uppy = useRef(Uppy().use(Transloadit, {}));
```

Comment:
This will create a "garbage" instance of Uppy on every render, because the `Uppy()` function is called even before `useRef()`--wonder if we can just create it once? :thinking:

Reply:
Thanks for reviewing it. I'll run some tests and come up with a solution.

Reply:
Did you find anything to solve this @pedrofs ?

Reply:
Hi @goto-bus-stop, I added a `useMemo` to prevent multiple initialisation of Uppy instance. I'm only thinking that once we are using `useMemo` we could get rid of `useRef`. What do you think here?